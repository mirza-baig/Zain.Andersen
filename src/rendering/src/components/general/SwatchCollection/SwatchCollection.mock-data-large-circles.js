const largeCirclesData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'SwatchCollection',
  },
  fields: {
    "componentSpacing": null,
    "sectionId": {
      "value": ""
    },
    "headlineLevel": {
      "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
      "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
      "name": "Heading 2",
      "displayName": "Heading 2",
      "fields": {
        "Value": {
          "value": "h2"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "headlineText": {
      "value": "Testing 2"
    },
    "swatchStyle": {
      "id": "f7015309-df16-46ad-8984-82799366def0",
      "url": "https://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Swatch-Style/Large-Circles",
      "name": "Large Circles",
      "displayName": "Large Circles",
      "fields": {
        "Value": {
          "value": "large-circles"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "swatchCollection": {
      "id": "9155f94b-820b-4971-9338-88ab00c64e11",
      "url": "/Data/Elements/Swatches/Collection/Entry-Doors-Exterior",
      "name": "Entry Doors Exterior",
      "displayName": "Entry Doors Exterior",
      "fields": {
        "swatchCollectionDescription": {
          "value": "Swatch collection Description Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book."
        },
        "swatchCollectionFooterCopy": {
          "value": "Swatch footer copy  - Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book."
        },
        "swatchCollectionName": {
          "value": "Big Doors Exterior"
        },
        "swatches": [
          {
            "id": "b60f6a67-ca34-4535-96a4-8f1c64199bc9",
            "url": "/Data/Elements/Swatches/Swatch/Interior/Red",
            "name": "Red",
            "displayName": "Red",
            "fields": {
              "swatchName": {
                "value": "Red this is a long name like a really really long name"
              },
              "swatchDescription": {
                "value": "#ffffff"
              },
              "swatchImage": {
                "value": {
                  "src": "https://preview.dev.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Swatches/cocoa-bean-exterior-color-swatch.png?h=77&iar=0&w=77&hash=E6992428AF7D7EE15D6A8134E6BBA10C",
                  "alt": "cocoa-bean-exterior-color-swatch",
                  "width": "77",
                  "height": "77"
                }
              }
            },
            "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
            "templateName": "Swatch"
          },
          {
            "id": "741493bb-2f08-4497-981f-b66c69eeead2",
            "url": "/Data/Elements/Swatches/Swatch/Interior/Blue",
            "name": "Blue",
            "displayName": "Blue",
            "fields": {
              "swatchName": {
                "value": "Blue"
              },
              "swatchDescription": {
                "value": ""
              },
              "swatchImage": {
                "value": {
                  "src": "https://preview.dev.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Swatches/dark-bronze-exterior-color-swatch.png?h=77&iar=0&w=77&hash=E2204D1CBCDDAEA254058BD9E3203DE6",
                  "alt": "dark-bronze-exterior-color-swatch",
                  "width": "77",
                  "height": "77"
                }
              }
            },
            "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
            "templateName": "Swatch"
          },
          {
            "id": "d42fb18b-74fb-4fc6-b9a4-7c175d3337bd",
            "url": "/Data/Elements/Swatches/Swatch/Interior/Green",
            "name": "Green",
            "displayName": "Green",
            "fields": {
              "swatchName": {
                "value": "Green"
              },
              "swatchDescription": {
                "value": "#ffffff"
              },
              "swatchImage": {
                "value": {
                  "src": "https://preview.dev.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Swatches/black-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=D12DE32E76267D6ED7F8A35D9D152127",
                  "alt": "black-exterior-80x80 (1)",
                  "width": "80",
                  "height": "80"
                }
              }
            },
            "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
            "templateName": "Swatch"
          }
        ]
      },
      "templateId": "c7e82176-d156-410f-b728-d4c465148578",
      "templateName": "Swatch Collection"
    },
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    },
    "children": []
  },};
export default largeCirclesData;
