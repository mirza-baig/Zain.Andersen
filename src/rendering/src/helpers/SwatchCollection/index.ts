export { default as SwatchCollection } from './SwatchCollection';
