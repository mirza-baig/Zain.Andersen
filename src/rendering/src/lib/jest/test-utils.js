// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
