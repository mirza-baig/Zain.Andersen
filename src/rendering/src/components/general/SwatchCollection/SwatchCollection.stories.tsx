// Global
import { Story, Meta } from '@storybook/react';
// Local
import SwatchCollection from './SwatchCollection';
import { SwatchCollectionProps } from './SwatchCollection';
import defaultData from './SwatchCollection.mock-data';
import largeCirclesData from './SwatchCollection.mock-data-large-circles';
import squaresData from './SwatchCollection.mock-data-squares';

export default {
  title: 'General/SwatchCollection',
  component: SwatchCollection,
} as Meta;
const Template: Story<SwatchCollectionProps> = (props) => <SwatchCollection {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
export const LargeCircles = Template.bind({});
LargeCircles.args = largeCirclesData;
export const Squares = Template.bind({});
Squares.args = squaresData;
