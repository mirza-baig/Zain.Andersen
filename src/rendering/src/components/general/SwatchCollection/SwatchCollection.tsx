// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import SwatchCollectionHelper from 'src/helpers/SwatchCollection/SwatchCollection';
import { Component } from 'src/helpers/Component';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';

export type SwatchCollectionProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.SwatchCollections.SwatchCollections;

const SwatchCollection = (props: SwatchCollectionProps) => {
  return (
    <Component variant="lg" dataComponent="general/swatchcollection" {...props}>
      <div className="col-span-12">
        <SwatchCollectionHelper layoutStyle="full-width" {...props} />
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<SwatchCollectionProps>(SwatchCollection);
