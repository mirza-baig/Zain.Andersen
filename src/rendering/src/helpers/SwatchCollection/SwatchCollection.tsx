import { getEnum } from 'lib/utils';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Headline } from 'src/helpers/Headline';
import ImageWrapper from 'src/helpers/Media/ImageWrapper';
import { BodyCopy } from 'src/helpers/BodyCopy';
import classNames from 'classnames';
import { useTheme } from 'lib/context/ThemeContext';
import { SwatchCollectionTheme } from './SwatchCollection.theme';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { Foundation } from 'src/.generated/Foundation.EnterpriseWeb.model';
import { EnumField } from 'lib/utils/get-enum';

export type SwatchStyles = 'small-circles' | 'large-circles' | 'square';
export type LayoutStyles = 'full-width' | 'side-by-side';

export type SwatchCollectionProps =
  Foundation.EnterpriseWeb.Enterprise.FieldSets.Products.SwatchCollections & {
    fields?: {
      swatchCollection?: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.SwatchCollection & {
        fields?: {
          swatches: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.Swatch[];
        };
      };
      swatchStyle?: EnumField<SwatchStyles>;
    };
    layoutStyle: LayoutStyles;
  };

const SwatchCollection = (props: SwatchCollectionProps) => {
  const swatchStyle = getEnum(props.fields?.swatchStyle) ?? 'large-circles';
  const { themeData, themeName } = useTheme(SwatchCollectionTheme(props.layoutStyle));

  let swatchImgClasses = '';
  let swatchesGap = '';
  let swatchWidth = '';

  switch (swatchStyle) {
    case 'small-circles':
      swatchImgClasses =
        'h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full';
      swatchWidth = 'w-[82px] ml:w-[86px] ';
      swatchesGap = classNames(
        'gap-y-m gap-x-[25px] ml:gap-x-[62px] ml:justify-start ml:pl-xxs justify-evenly',
        themeName === 'rba' ? 'ml:pl-m mt-m ml:mt-l mb-s ml:gap-y-l' : 'ml:gap-y-m ml:pl-0'
      );
      break;

    case 'large-circles':
      swatchImgClasses = 'h-xxl w-xxl ml:h-[160px] ml:w-[160px] rounded-full [&_img]:rounded-full';
      swatchWidth = 'w-xxl ml:w-[160px] ml:pl-0';
      swatchesGap = classNames(
        'gap-x-[52px] gap-y-m ml:gap-x-[38px] ml:justify-start justify-evenly ml:pl-0',
        themeName === 'rba' ? 'my-s' : ''
      );
      break;

    case 'square':
      swatchImgClasses = 'h-[156px] w-[156px] ml:h-[160px] ml:w-[160px]';
      swatchWidth = 'w-[156px] ml:w-[160px] ';
      swatchesGap = classNames(
        'gap-x-[15px] gap-y-m ml:justify-start justify-evenly',
        themeName === 'rba' ? 'mt-m ml:gap-x-l' : 'ml:gap-x-[43px]'
      );
      break;
  }

  return (
    <div>
      <Headline
        useTag="h3"
        classes={themeData.classes.swatchTitle}
        fields={{
          headlineText: props.fields?.swatchCollection?.fields.swatchCollectionName ?? {
            value: '',
          },
        }}
      />

      <BodyCopy
        classes={themeData.classes.swatchDescription}
        fields={{
          body: props.fields?.swatchCollection?.fields.swatchCollectionDescription ?? { value: '' },
        }}
      />

      <div className={classNames('flex flex-wrap [&>*:last-child]:justify-start', swatchesGap)}>
        {props.fields?.swatchCollection?.fields.swatches.map(
          (swatch: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.Swatch, index: number) => {
            return (
              <div key={index} className={classNames(swatchWidth, 'flex flex-col')}>
                <div className={classNames(swatchImgClasses, 'mx-auto mb-xxs')}>
                  <ImageWrapper imageLayout="responsive" image={swatch.fields.swatchImage} />
                </div>
                <Headline
                  useTag="h4"
                  classes={themeData.classes.swatchLabel}
                  fields={{
                    headlineText: swatch.fields.swatchName ?? {
                      value: '',
                    },
                  }}
                />
              </div>
            );
          }
        )}
      </div>
      <RichTextWrapper
        classes={themeData.classes.swatchFooterCopy}
        field={props.fields?.swatchCollection?.fields.swatchCollectionFooterCopy ?? { value: '' }}
      />
    </div>
  );
};

export default SwatchCollection;
