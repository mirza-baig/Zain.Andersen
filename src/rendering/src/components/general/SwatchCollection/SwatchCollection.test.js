// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import SwatchCollection from './SwatchCollection';
import defaultData from './SwatchCollection.mock-data';

it('renders correctly', async () => {
  const component = snapshot(SwatchCollection, { componentProps: defaultData });
  await hasDataComponent(component, 'general/swatchcollection');
});
