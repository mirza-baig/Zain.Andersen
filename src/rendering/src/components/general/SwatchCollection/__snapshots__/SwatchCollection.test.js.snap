// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/swatchcollection"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div>
          <div
            class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-xs ml:text-xs font-medium mt-m ml:mt-l  mb-xs"
          >
            <h3>
              Big Doors Exterior
            </h3>
          </div>
          <div
            class="font-regular text-dark-gray text-body mb-m"
          >
            <div
              class="body-copy"
            >
              <div>
                Swatch collection Description Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
              </div>
            </div>
          </div>
          <div
            class="flex flex-wrap [&>*:last-child]:justify-start gap-y-m gap-x-[25px] ml:gap-x-[62px] ml:justify-start ml:pl-xxs justify-evenly ml:gap-y-m ml:pl-0"
          >
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="black-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="black-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  this is a long name like a really really long name
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="cocoa-bean-exterior-color-swatch"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="cocoa-bean-exterior-color-swatch"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Black
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="sandtone-exterior-color-swatch"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="sandtone-exterior-color-swatch"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Sandstone
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="terratone-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="terratone-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Orange
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="white-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="white-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Black
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="terratone-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="terratone-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Black
                </h4>
              </div>
            </div>
            <div
              class="w-[82px] ml:w-[86px]  flex flex-col"
            >
              <div
                class="h-[64px] ml:h-[86px] w-[64px] ml:w-[86px] rounded-full [&_img]:rounded-full mx-auto mb-xxs"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="white-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="white-exterior-80x80 (1)"
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="items-top font-sans [&_a:hover]:underline mx-auto !font-serif text-dark-gray text-small text-center font-regular "
              >
                <h4>
                  Black
                </h4>
              </div>
            </div>
          </div>
          <div
            class="body-copy font-regular text-theme-body text-small text-dark-gray mt-m"
          >
            <div>
              Swatch footer copy  - Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
